tests/test_dockerspawner.py

```diff
@@ -82,27 +82,48 @@ async def test_start_stop(dockerspawner_configured_app, remove):
         assert sorted(state) == ["object_id", "object_name"]
 
 
-@pytest.mark.parametrize("image", ["1.0", "1.1.0", "nomatch"])
-async def test_allowed_image(dockerspawner_configured_app, image):
+def allowed_images_callable(*_):
+    return ["jupyterhub/singleuser:1.0", "jupyterhub/singleuser:1.1"]
+
+
+@pytest.mark.parametrize(
+    "image, expected",
```

Comment:
Let's call the variable what it sets, to make it clearer what it is:

- `allowed_images` is the allowed_images configuration
- `image` is the requested image

```suggestion
    "allowed_images, image",
```